src/common/utxobased/db/makeProcessor.ts

```diff
@@ -78,7 +71,7 @@ export interface Processor {
   removeUtxos: (utxoIds: string[]) => Promise<void>
   // fetch either all UTXOs if the array is empty or as selected from an array
   // of UTXO ids
-  fetchUtxos: (args: FetchUtxosArgs) => Promise<IUTXO[]>
+  fetchUtxos: (args: FetchUtxosArgs) => Promise<Array<IUTXO | undefined>>
```

Comment:
How can you have "undefined"?

Reply:
If the query parameters include a UTXO ID that doesn't exist in the database, then undefined is returned for that entry:

```js
fetchUtxo({ utxoIds: ['x', 'y'] }) // Returns [x, undefined] if 'y' ID does not exist.
```